File: competitive_programming/leetcode/06_javascript/412_fizz_buzz.js
function fizzBuzz(n) {
  const result = [];
  for (let value = 1; value <= n; value += 1) {
    if (value % 15 === 0) result.push("FizzBuzz");
    else if (value % 3 === 0) result.push("Fizz");
    else if (value % 5 === 0) result.push("Buzz");
    else result.push(String(value));
  }
  return result;
}

if (require.main === module) {
  if (fizzBuzz(5).join(",") !== "1,2,Fizz,4,Buzz") throw new Error("fizz buzz failed");
  if (fizzBuzz(15)[14] !== "FizzBuzz") throw new Error("fizz buzz failed");
  console.log("412 fizz buzz ok");
}

File: competitive_programming/leetcode/11_golang/412_fizz_buzz.go
package main

import "fmt"

func fizzBuzz(n int) []string {
	result := []string{}
	for value := 1; value <= n; value++ {
		switch {
		case value%15 == 0:
			result = append(result, "FizzBuzz")
		case value%3 == 0:
			result = append(result, "Fizz")
		case value%5 == 0:
			result = append(result, "Buzz")
		default:
			result = append(result, fmt.Sprintf("%d", value))
		}
	}
	return result
}

func main() {
	if fmt.Sprint(fizzBuzz(5)) != "[1 2 Fizz 4 Buzz]" || fizzBuzz(15)[14] != "FizzBuzz" {
		panic("fizz buzz failed")
	}
	fmt.Println("412 fizz buzz ok")
}
